src/field/FieldDriver.hh

```diff
@@ -153,10 +153,10 @@ FieldDriver<StepperT>::advance(real_type step, const OdeState& state)
 
     if (rel_error > 1)
     {
-        // Advance more accurately with a newly proposed step
+        // Discard the original end state and advance more accurately with the
+        // newly proposed step
         real_type next_step = this->new_step_size(step, rel_error);
-        output.end          = this->accurate_advance(
-            output.end.step, output.end.state, next_step);
+        output.end = this->accurate_advance(output.end.step, state, next_step);
```

Comment:
Good catch!  I feel like I should retire now.

Reply:
For the night yes! For your career, definitely not!

Reply:
Also, this was *your* catch, not mine! The commit has you as the author :) because you found this in January.

Reply:
Another good reason to retire as I do not remember that!